framework/i18n/Formatter.php

```diff
@@ -828,18 +828,75 @@ public function asRelativeTime($value, $referenceTime = null)
      * @return string the formatted duration.
      * @since 2.0.7
+     * @deprecated Since version 2.0.10. The signature will be changed to [[Formatter::asDurationTime()]] in 2.1
+     * @see asDurationTime()
      */
     public function asDuration($value, $implodeString = ', ', $negativeSign = '-')
     {
-        if ($value === null) {
+        return $this->asDurationTime($value, ['implodeString' => $implodeString, 'negativeSign' => $negativeSign]);
+    }
+
+    /**
+     * Represents the value as duration in human readable format.
+     * @param DateInterval|string|integer $value the value to be formatted. Acceptable formats:
+     *  - [DateInterval object](http://php.net/manual/ru/class.dateinterval.php)
+     *  - integer - number of seconds. For example: value `131` represents `2 minutes, 11 seconds`
+     *  - ISO8601 duration format. For example, all of these values represent `1 day, 2 hours, 30 minutes` duration:
+     *    `2015-01-01T13:00:00Z/2015-01-02T13:30:00Z` - between two datetime values
+     *    `2015-01-01T13:00:00Z/P1D2H30M` - time interval after datetime value
+     *    `P1D2H30M/2015-01-02T13:30:00Z` - time interval before datetime value
+     *    `P1D2H30M` - simply a date interval
+     *    `P-1D2H30M` - a negative date interval (`-1 day, 2 hours, 30 minutes`)
+     * @param array $options The parameter of human readable format string. Includes:
+     *    - `string` `implodeString` will be used to concatenate duration parts. Defaults to `, `.
+     *    - `string` `negativeSign` will be prefixed to the formatted duration, when it is negative. Defaults to `-`.
+     *    - `null|true|false` `showSeconds` will be displayed  part for seconds data. Default is null that means is
+     *      seconds data is displayed if not equals 0 value. To set `true` if you want display any value of parts always.
+     *      False - never display a part.
+     *    - `null|true|false` `showMinutes` will be displayed part for minutes data. Behaves like as `showSeconds`.
+     *    - `null|true|false` `showHours` will be displayed part for hours data. Behaves like as `showSeconds`.
+     *    - `null|true|false` `showDays` will be displayed part for days data. Behaves like as `showSeconds`.
+     *    - `null|true|false` `showMonths` will be displayed part for months data. Behaves like as `showSeconds`.
+     *    - `null|true|false` `showYears` will be displayed part for years data. Behaves like as `showSeconds`.
+     * Note that if `value` parameter equals zero value, will be displayed `0 seconds`, if `showSeconds` not `false`.
+     * And if `showYears` is `false` that is mean low priority parts(`showMonths`, `showDays`, `showDays`, ...) will be
+     * set to false also.
+     * @return string the formatted duration.
+     * @since 2.0.10
+     */
+    public function asDurationTime($value, $options = [])
+    {
+        if (is_null($value)) {
```

Comment:
It's overall better to use simple checks instead of function calls where possible. More explicit, less opcodes.